packages/react-components/react-select/src/components/Select/useSelectStyles.ts

```diff
@@ -108,25 +148,45 @@ const useSelectStyles = makeStyles({
       ...shorthands.borderColor('GrayText'),
     },
   },
+
   small: {
     height: fieldHeights.small,
-    ...shorthands.padding('0', tokens.spacingHorizontalSNudge),
+    paddingBottom: 0,
+    paddingLeft: 'var(--selectSpacingLeftS)',
```

Comment:
This is just a suggestion, feel free to disregard. What about taking CSS vars one step further and just having one for left and one for right and then conditionally setting the values?

It would look roughly like:

```ts
makeStyles({
  base: {
    paddingLeft: 'var(--selectSpacingLeft)',
    paddingRight: 'var(--selectSpacingRight)'
  },
  small: {
    '--selectSpacingLeft': paddingLeft.small,
    '--selectSpacingRight': paddingRight.small
  },
  medium: {
    '--selectSpacingLeft': paddingLeft.medium,
    '--selectSpacingRight': paddingRight.medium
  }
  // ....
})
```

I think this is a bit simpler because there are really only two variables and their values change based on the size rather than having six variables. It will also eliminate a couple lines of code :)

[Here's a working Codesandbox example](https://codesandbox.io/s/css-vars-8ody0h?file=/src/index.tsx).

Reply:
The reason I didn't do that is because I thought having separate variables would make it easier for someone who wants to override the spacing across all sizes, just by manipulating variables. I suppose it depends on how likely you think it is for app authors to be mixing sizes within the same app, while also wanting to override their spacing styles 😄

Reply:
So the case is that:
1. There are Selects of multiple sizes on the page
2. Authors want to modify the spacing on _just_ the medium sizes Selects across the page but leave the smalls and larges alone?

Reply:
Was thinking about this some more and if we want to support overriding CSS vars we should consider those vars as part of the component API. I know that we cannot technically prevent users for overriding the CSS vars but I think we should consider them private, internal implementation details.

The spacing values are based on spacing values from the theme so the correct way to change them is to change the theme imo. I realize that targeting the CSS vars provides finer-grained control over the spacing here as an author can change just the spacing for Select rather than the entire theme but nested ThemeProviders can address that and I don't know that we need this feature right now so I'm in favor of not going out of our way to support it at this time. We can always add it later. Do you know of a case where we need to support this scenario?